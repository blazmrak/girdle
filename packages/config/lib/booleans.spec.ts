import { initializeEnvironment, Var } from './index'
import { expectTypeOf } from 'expect-type'

describe('boolean', () => {
  it('parses basic options', () => {
    const env = { BOOL: 'true', OD: 'false' }

    const config = initializeEnvironment(
      {
        defaultBool: Var.boolean().default(true),
        od: Var.boolean().default(true),
        optionalBool: Var.boolean().optional(),
        bool: Var.boolean(),
        namedBool: Var.boolean().name('BOOL'),
      },
      { env },
    )

    expect(config).toMatchSnapshot()
    expectTypeOf(config).toEqualTypeOf<{
      defaultBool: boolean
      od: boolean
      optionalBool: boolean | undefined
      bool: boolean
      namedBool: boolean
    }>()
  })
})
