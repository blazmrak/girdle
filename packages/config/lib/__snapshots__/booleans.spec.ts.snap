// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boolean parses basic options 1`] = `
{
  "bool": true,
  "defaultBool": true,
  "namedBool": true,
  "od": false,
  "optionalBool": undefined,
}
`;
